nixos/modules/services/logging/logrotate.nix

```diff
@@ -82,27 +83,67 @@ let
         '';
       };
     };
-
-    config.name = name;
   };
 
+  generateLine = k: v:
+    # special keys handled separately for backwards compatibility
+    if builtins.elem k ["name" "path" "priority" "enable" "user" "group" "keep"] then ""
+    else if k == "extraConfig" then "${v}"
+    else if k == "frequency" then "${v}\n"
+    else if k == "keep" then "rotate ${toString v}\n"
+    else if builtins.elem k ["firstaction" "lastaction" "prerotate" "postrotate" "preremove"]
+         then "${k}\n  ${v}\nendscript\n"
+    else if v == null then ""
+    else if isInt v then "${k} ${toString v}\n"
+    else if v == true then "${k}\n"
+    else if v == false then "no${k}\n"
+    else "${k} ${v}\n";
+  generateSection = settings: concatStringsSep "" (
+    (mapAttrsToList generateLine settings)
+    ++ [
+        # user and group must be handled together
+        (if (settings.user or null != null && settings.group or null != null)
```

Comment:
Since `user` and `group` have default values this should be equivalent to `settings.user != null` and `settings.group != null`. If you ever need to see if a module option has been defined, you may add `options` to the `{ config, lib, pkgs, ... }:` and use `options.services.logrotate.settings.foo.isDefined`.

Reply:
Yes and no, that would be true for `paths` attrsets but `generateSection` is also used with `services.logrotate.settings` which doesn't have default values there.